### DISCUSSION TITLE
Setting permissions for created certs and directories

### DISCUSSION THREAD

**Discussion by arisudesu** [go-acme/lego#1822] [Q&A]
Is it possible to set permissions for files and folders that lego creates in its work?

What I am trying to achieve is to place lego's data to `/etc/lego` owned by root, then share this directory with group `certusers` to give access to certificates to other processes, which aren't run as root. It is possible to copy directory gid to nested files and dirs by using `chmod g+s /etc/lego`, however they won't be readable by certusers because those files have no group permissions.

**Comment by ldez** [comment]
Hello,

you can use `--run-hook` or `--renew-hook` to add a script to handle permissions.

**Reply by arisudesu** [reply]
Sure I can, I wanted to avoid adding a hook for this purpose though. Hook is extra step and requires having extra scripts on the disk. Maybe it's possible to add umask support to lego?
